tests/generation/UtilitiesTest.py

```diff
@@ -73,11 +77,89 @@ def test_get_all_args(self):
                           'no_chars': 79,
                           'chars': False, 'dependency_results': {}})
 
+
+class TestHashBang(unittest.TestCase):
+
+    def test_missing_file(self):
+        self.assertIsNone(get_hashbang('does_not_exist'))
+
+    def test_with_bash(self):
+        with NamedTemporaryFile(mode='w+t', delete=False) as temp_file:
+            temp_file.write('#!bin/bash\n')
+            temp_file.close()
+            self.assertEqual(get_hashbang(temp_file.name), '#!bin/bash')
+
+    def test_no_eol(self):
+        with NamedTemporaryFile(mode='w+t', delete=False) as temp_file:
+            temp_file.write('#!bin/bash')
+            temp_file.close()
+            self.assertEqual(get_hashbang(temp_file.name), '#!bin/bash')
+
+    def test_with_slash(self):
+        with NamedTemporaryFile(mode='w+t', delete=False) as temp_file:
+            temp_file.write('#!/bin/bash\n')
+            temp_file.close()
+            self.assertEqual(get_hashbang(temp_file.name), '#!/bin/bash')
+
+    def test_with_space(self):
+        with NamedTemporaryFile(mode='w+t', delete=False) as temp_file:
+            temp_file.write('#!/bin/bash \n')
+            temp_file.close()
+            self.assertEqual(get_hashbang(temp_file.name), '#!/bin/bash')
+
+    def test_env(self):
+        with NamedTemporaryFile(mode='w+t', delete=False) as temp_file:
+            temp_file.write('#!/bin/env bash\n')
+            temp_file.close()
+            self.assertEqual(get_hashbang(temp_file.name), '#!/bin/env bash')
+
+    def test_non_unicode_file(self):
+        with NamedTemporaryFile(mode='w+b', delete=False) as temp_file:
+            temp_file.write(b'\2000x80')
+            temp_file.close()
+            self.assertIsNone(get_hashbang(temp_file.name))
+
+    def test_empty_file(self):
+        with NamedTemporaryFile(mode='w+t', delete=False) as temp_file:
+            temp_file.write('\n')
+            temp_file.close()
+            self.assertIsNone(get_hashbang(temp_file.name))
+
+    def test_no_bang(self):
+        with NamedTemporaryFile(mode='w+t', delete=False) as temp_file:
+            temp_file.write('#bin/bash')
+            temp_file.close()
+            self.assertIsNone(get_hashbang(temp_file.name))
+
+    def test_no_hash(self):
+        with NamedTemporaryFile(mode='w+t', delete=False) as temp_file:
+            temp_file.write('!bin/bash')
+            temp_file.close()
+            self.assertIsNone(get_hashbang(temp_file.name))
+
     def test_get_language_from_hashbang(self):
         self.assertEqual(get_language_from_hashbang('#!/usr/bin/env python'),
                          'python')
         self.assertEqual(get_language_from_hashbang('#!bin/bash'),
                          'bash')
+        self.assertEqual(get_language_from_hashbang('#!/bin/bash'),
+                         'bash')
+
+    def test_split_by_language(self):
+        with NamedTemporaryFile(delete=False, suffix='.py') as temp_file1, \
```

Comment:
is the trailing backslash necessary over here?

Reply:
it doesn't seem like it **is** necessary

Reply:
actually this is a oddity with the language -- multipe context handlers in a single `if` must have ` \` -- brackets are a syntax error.

Reply:
@ishanSrt , Are you asking me to change it to nested `with`?

Reply:
If everything is fine, you need to approve the PR.